assemblies/assembly-planner-java.adoc

```diff
@@ -0,0 +1,750 @@
+include::{enterprise-dir}/optimizer/business-optimizer-logging-quarkus-proc.adoc[leveloffset=+1]
+
+== Using Micrometer and Prometheus to monitor your school timetable OptaPlanner Java application
+
+OptaPlanner exposes metrics through https://micrometer.io/[Micrometer], a metrics instrumentation library for Java applications. You can use Micrometer with Prometheus to monitor the OptaPlanner solver in the school timetable application.
+
+To configuring a plain Java OptaPlanner application to use Micrometer, you must
+
+
+.Prerequisites
+* You have created the OptaPlanner school timetable application with Java.
+
+.Procedure
+. Add the Micrometer Promtheus dependencies to the school timetable `pom.xml` file:
++
+[source]
+----
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-registry-prometheus</artifactId>
+ <version>1.7.3</version>
+</dependency>
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-core</artifactId>
+ <version>1.7.3</version>
+</dependency>
+----
+
+. Add the following lines to the top of the `pom.xml` file so Prometheus can scrap data from `com.sun.net.httpserver.HttpServer`  before the solutions starts:
++
+[source,java,nowrap]
+----
+PrometheusMeterRegistry prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+try {
+    HttpServer server = HttpServer.create(new InetSocketAddress(8080), 0);
+    server.createContext("/prometheus", httpExchange -> {
+        String response = prometheusRegistry.scrape();
+        httpExchange.sendResponseHeaders(200, response.getBytes().length);
+        try (OutputStream os = httpExchange.getResponseBody()) {
+            os.write(response.getBytes());
+        }
+    });
+    new Thread(server::start).start();
+} catch (IOException e) {
+    throw new RuntimeException(e);
+}
+Metrics.addRegistry(prometheusRegistry);
+----
+
+. Add the following line to control the solving time. This will let you see how the metrics change based on the time spent solving.
++
+[source]
+----
+withTerminationSpentLimit(Duration.ofMinutes(5)));
+----
+. Start the school timetable application.
+. Open `http://localhost:8080/prometheus` in a web browser to view the timetable application in Prometheus.
+. Open your monitoring system to view the metrics for your OptaPlanner project.
++
+The following metrics are exposed:
++
+* `optaplanner_solver_errors_total`: the total number of errors that occurred while solving since the start
+of the measuring.
+* `optaplanner_solver_solve_length_active_count`: the number of solvers currently solving.
+* `optaplanner_solver_solve_length_seconds_max`: run time of the
+longest-running currently active solver.
+* `optaplanner_solver_solve_length_seconds_duration_sum`: the sum of each active solver's solve duration. For example, if there are two active solvers, one running for three minutes and the other for one minute, the total solve time is four minutes.
```

Comment:
As mentioned here https://kie.zulipchat.com/#narrow/stream/235756-optaplanner-dev/topic/Micrometer.3A.20solve-length.20metric.20name/near/253046064
Names of metrics should be changed to 
optaplanner_solver_solve_length_seconds_max -> optaplanner_solver_solve_duration_seconds_max
optaplanner_solver_solve_length_seconds_active_count -> optaplanner_solver_solve_duration_seconds_active_count
optaplanner_solver_solve_length_seconds_duration_sum -> optaplanner_solver_solve_duration_seconds_duration_sum

Reply:
OK. I didn't the middle one. Please check to so if my changes are correct.

Reply:
Sorry maybe I am blind I do not see the change of the word length to duration

Reply:
Look on this page:
https://github.com/emmurphy1/kie-docs/pull/336/files
You might have to show the diffs. The file is assembly-planner-java

Reply:
LGTM